pact/brackets-review.pact

```diff
@@ -0,0 +1,504 @@
+
+
+(module brackets BRACKETS-GOVERNANCE
+
+  (defun enforce-brackets-admin ()
+    (enforce-guard (read-keyset "master-bracket-keyset"))
+  )
+
+  (defcap BRACKETS-GOVERNANCE ()
+    (enforce-brackets-admin)
+  )
+
+  (use coin)
+
+  (defconst CONTRACT_ACCOUNT:string 'master-bracket-keyset)
+  (defun contract-guard:guard () (create-module-guard 'master-bracket-keyset))
+
+   ;game initiated, people still signing up
+   (defconst INITIATED:string "initiated")
+   ;people signed up, game in progress
+   (defconst IN_PROGRESS:string "in-progress")
+   ;game finished and winner not paid
+   (defconst COMPLETE:string "complete")
+   ;game finished and winner paid
+   (defconst WINNER_PAID:string "winner-paid")
+   ;team has no player assigned
+   (defconst UNASSIGNED:string "unassigned")
+
+  (defschema empty-bracket
+    ;will need to refactor admin to take in a keyset
+    ;not sure how to do this from the frontend though...
+    admin:string
+    bracket:list
+    status:string
+    moneyPool:integer
+    players:[string]
+    winner:string
+    entryFee:integer
+    usernames:[string])
+    ;guard:guard
+
+  (deftable empty-bracket-table:{empty-bracket})
+
+  (defschema bracket-betting
+    ;will need to refactor admin to take in a keyset
+    ;not sure how to do this from the frontend though...
+    admin:string
+    bracket:list
+    status:string
+    moneyPool:integer
+    players:[string]
+    players-bets:list
+    winner:string
+    entryFee:integer)
+
+  (deftable bracket-betting-table:{bracket-betting})
+
+  (defschema users
+    username:string
+    user-guard:guard
+    balance:integer
+    bb-games:list
+    bb-admins:list
+    eb-games:list
+    eb-admins:list
+    games-won:list)
+
+  (deftable users-table:{users})
+
+  (defcap BRACKET-ADMIN-BB (admin-key:string bracket-name:string)
```

Comment:
A capability represents some unforgeable fact, usually by enforcing a keyset. See comments regarding inserting a guard in for the bracket admin. Once that is in, this capability should read the guard from the table and `enforce-guard` on it. As it is this is simply string equality which is easy to spoof. Yes, "sender" will fail if you are not the proper account, but gas spender vs admin identity are bad things to confuse.